### DISCUSSION TITLE
Type of the Query param not shown in /docs

### DISCUSSION THREAD

**Discussion by looprewind** [fastapi/fastapi#10781] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI, Query

app = FastAPI()


@app.get("/items/")
async def read_items(q: Annotated[str | None, Query()] = None):
    query_items = {"q": q}
    return query_items
```

### Description

Hi, since v.0.103 I have been noticing that query params that uses the Annotated typing, does not display the type of the param in the /docs. On /redoc works flawlessly.

![Capture1](https://github.com/tiangolo/fastapi/assets/2674858/32ab7ab2-7d86-42e6-a6a0-fb18c349d858)
![Capture2](https://github.com/tiangolo/fastapi/assets/2674858/61b55395-94c1-443b-b176-0eb1abb9a5d8)

I also have tried the [example shown on the documentation](https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#query-parameter-list-multiple-values) and the type of the query paramer is not shown on /docs. Good news are that if you declare the param like `q: str = Query()` is correctly shown on /docs.

I tried to detect if the error is on FastAPI, Pydantic or even Swagger UI but I can't see were the type of a parameter is passed, in my opinion some knowledge about this will enrich the discussion!

Thanks a lot!

### Operating System

Windows

### FastAPI Version

0.105.0

### Pydantic Version

2.42

### Python Version

3.11

**Comment by dmontagu** [comment, marked as answer]
So, what you see in `/docs` or `/redoc` is going to be based on how those pages choose to render the OpenAPI schema.

As far as I can tell, you get the same OpenAPI spec with or without annotations:
```python
from typing import Annotated

from fastapi import FastAPI, Query

app = FastAPI()


@app.get("/items-1/")
async def read_items_1(q: str | None = Query(None)):
    query_items = {"q": q}
    return query_items


@app.get("/items-2/")
async def read_items_2(q: Annotated[str | None, Query()] = None):
    query_items = {"q": q}
    return query_items


print(app.openapi()['paths']['/items-1/']['get']['parameters'])
#> [{'name': 'q', 'in': 'query', 'required': False, 'schema': {'anyOf': [{'type': 'string'}, {'type': 'null'}], 'title': 'Q'}}]
print(app.openapi()['paths']['/items-2/']['get']['parameters'])
#> [{'name': 'q', 'in': 'query', 'required': False, 'schema': {'anyOf': [{'type': 'string'}, {'type': 'null'}], 'title': 'Q'}}]
```
(Generated with fastapi v0.103.2)

So I don't think this has to do with the use of annotations.

It looks like the default Swagger-UI version was updated in fastapi 0.99.0 and in 0.104.1. (You want to look at the default value of `swagger_js_url` in `fastapi.openapi.docs.get_swagger_ui_html`.)

If you notice a difference in behavior in `fastapi<0.99.0` or `fastapi<0.104.1` that would probably explain it.

**Reply by looprewind** [reply]
Indeed! The problem relies on the swagger-ui and the way that is rendering the openapi schema. Thanks a lot for the answer @dmontagu ;)
